internal/executor/executor.go

```diff
@@ -75,21 +75,23 @@ func NewExecutor(
 }
 
 func (executor *Executor) RunBuild() {
-	initialStepsRequest := api.InitialCommandsRequest{
+	log.Println("Getting initial commands...")
+	response, err := client.CirrusClient.InitialCommands(context.Background(), &api.InitialCommandsRequest{
 		TaskIdentification:  executor.taskIdentification,
 		LocalTimestamp:      time.Now().Unix(),
 		ContinueFromCommand: executor.commandFrom,
-	}
-	log.Println("Getting initial commands...")
-	response, err := client.CirrusClient.InitialCommands(context.Background(), &initialStepsRequest)
+	})
+	retryDelay := 5 * time.Second
 	for response == nil || err != nil {
-		request := api.ReportAgentProblemRequest{
-			TaskIdentification: executor.taskIdentification,
-			Message:            fmt.Sprintf("Failed to get initial commands: %v", err),
-		}
-		client.CirrusClient.ReportAgentWarning(context.Background(), &request)
-		time.Sleep(10 * time.Second)
-		response, err = client.CirrusClient.InitialCommands(context.Background(), &initialStepsRequest)
+		log.Printf("Failed to get initial commands: %v", err)
+		retryDelay *= 2
```

Comment:
Consider using go-retry instead to avoid re-implementing things like exponential backoff.

Reply:
`github.com/avast/retry-go` that is using in CLI doesn't seem to provide an option to modify the function to be retried on the first retry. In this case to modify the request to include `Retry` flag.